src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +266,252 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
+                            objectTypes.TryGetValue(targetEntityName, out ObjectTypeDefinitionNode? targetObjectTypeDefinitionNode))
+                        {
+                            IReadOnlyList<FieldDefinitionNode> gg = targetObjectTypeDefinitionNode.Fields;
+                            Dictionary<string, FieldDefinitionNode> targetFieldDefinitions = targetObjectTypeDefinitionNode.Fields.ToDictionary(field => field.Name.Value, field => field);
+                            // When target entity is the referencing entity, referencing field directive is to be added to relationship fields
+                            // in the target entity.
+                            AddReferencingFieldDirectiveToReferencingFields(targetFieldDefinitions, targetReferencingFKInfo.ReferencingColumns, sqlMetadataProvider, targetEntityName);
+
+                            // Update the target object definition with the new set of fields having referencing field directive.
+                            objectTypes[targetEntityName] = targetObjectTypeDefinitionNode.WithFields(new List<FieldDefinitionNode>(targetFieldDefinitions.Values));
+                        }
+                    }
+                }
+
+                // Update the source object definition with the new set of fields having referencing field directive.
+                objectTypes[sourceEntityName] = sourceObjectTypeDefinitionNode.WithFields(new List<FieldDefinitionNode>(sourceFieldDefinitions.Values));
+            }
+        }
+
+        /// <summary>
+        /// Helper method to add referencing field directive type to all the fields in the entity which
+        /// hold a foreign key reference to another entity exposed in the config, related via a relationship.
+        /// </summary>
+        /// <param name="referencingEntityFieldDefinitions">Field definitions of the referencing entity.</param>
+        /// <param name="referencingColumns">Referencing columns in the relationship.</param>
+        private static void AddReferencingFieldDirectiveToReferencingFields(
+            Dictionary<string, FieldDefinitionNode> referencingEntityFieldDefinitions,
+            List<string> referencingColumns,
+            ISqlMetadataProvider metadataProvider,
+            string entityName)
+        {
+            foreach (string referencingColumn in referencingColumns)
+            {
+                if (metadataProvider.TryGetExposedColumnName(entityName, referencingColumn, out string? exposedReferencingColumnName) &&
+                    referencingEntityFieldDefinitions.TryGetValue(exposedReferencingColumnName, out FieldDefinitionNode? referencingFieldDefinition))
+                {
+                    if (!referencingFieldDefinition.Directives.Any(directive => directive.Name.Value == ReferencingFieldDirectiveType.DirectiveName))
+                    {
+                        List<DirectiveNode> directiveNodes = referencingFieldDefinition.Directives.ToList();
```

Comment:
Can't we simply add to the existing `Directives` object? Do we need to first convert into a new list?

Reply:
We cannot. `Directives` return an `IReadOnlyList`.